drivers/vmwareappcatalyst/appcatalyst_darwin.go

```diff
@@ -0,0 +1,623 @@
+	if d.SSHPort == 0 {
+		d.SSHPort = 22
+	}
+
+	return d.SSHPort, nil
+}
+
+func (d *Driver) GetSSHUsername() string {
+	if d.SSHUser == "" {
+		d.SSHUser = "photon"
+	}
+
+	return d.SSHUser
+}
+
+func (d *Driver) DriverName() string {
+	return "vmwareappcatalyst"
+}
+
+func (d *Driver) SetConfigFromFlags(flags drivers.DriverOptions) error {
+	d.Memory = flags.Int("vmwareappcatalyst-memory-size")
+	d.CPU = flags.Int("vmwareappcatalyst-cpu-count")
+	d.SwarmMaster = flags.Bool("swarm-master")
+	d.SwarmHost = flags.String("swarm-host")
+	d.SwarmDiscovery = flags.String("swarm-discovery")
+	d.SSHUser = "photon"
+	d.SSHPort = 22
+	d.VMLocation = flags.String("vmwareappcatalyst-vm-location")
+	d.APIPort = flags.Int("vmwareappcatalyst-api-port")
+
+	// We support a maximum of 16 cpu.
+	if d.CPU > 16 {
+		d.CPU = 16
+	}
+
+	return nil
+}
+
+func (d *Driver) GetURL() (string, error) {
+	ip, err := d.GetIP()
+	if err != nil {
+		return "", err
+	}
+	if ip == "" {
+		return "", nil
+	}
+	return fmt.Sprintf("tcp://%s:2376", ip), nil
+}
+
+func (d *Driver) GetIP() (string, error) {
+	s, err := d.GetState()
+	if err != nil {
+		return "", err
+	}
+	if s != state.Running {
+		return "", drivers.ErrHostIsNotRunning
+	}
+
+	c, err := NewClient("http://localhost:" + strconv.Itoa(d.APIPort))
+	if err != nil {
+		return "", err
+	}
+
+	var ip string
+	for i := 1; i <= 60; i++ {
+		vmip, err := c.GetVMIPAddress(d.MachineName)
+		if err != nil && vmip.Code != 200 {
+			log.Debugf("Not there yet %d/%d, error code: %d, message %s", i, 60, vmip.Code, err)
+			time.Sleep(2 * time.Second)
+			continue
+		}
+		ip = vmip.Message
+		log.Debugf("Got an ip: %s", ip)
+		break
+	}
+
+	if ip == "" {
+		return "", fmt.Errorf("machine didn't return an IP after 120 seconds, aborting")
+	}
+
+	return ip, nil
+}
+
+func (d *Driver) GetState() (state.State, error) {
+	c, err := NewClient("http://localhost:" + strconv.Itoa(d.APIPort))
+	if err != nil {
+		return state.Error, err
+	}
+
+	power, err := c.GetPowerVM(d.MachineName)
+	if err != nil {
+		return state.Error, err
+	}
+
+	switch power.Message {
+	case "powering on":
+		return state.Running, nil
+	case "powered on":
+		return state.Running, nil
+	case "powering off":
+		return state.Stopped, nil
+	case "powered off":
+		return state.Stopped, nil
+	case "reseting":
+		return state.Stopped, nil
+	case "suspended":
+		return state.Stopped, nil
+	case "suspending":
+		return state.Stopped, nil
+	case "tools_running":
+		return state.Running, nil
+	case "blocked on msg":
+		return state.Error, nil
+	}
+
+	// No state whatsoever, errors out
+	return state.Error, nil
+}
+
+func (d *Driver) PreCreateCheck() error {
+	return nil
+}
+
+func (d *Driver) Create() error {
+
+	log.Infof("Creating SSH key...")
+	if err := ssh.GenerateSSHKey(d.GetSSHKeyPath()); err != nil {
+		return err
+	}
+
+	c, err := NewClient("http://localhost:" + strconv.Itoa(d.APIPort))
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Creating VM...")
```

Comment:
I'm assuming that AppCatalyst takes care of creating this VM upon installation?  Will this work with a non-Photon based OS?